s4e-web/src/app/views/login/login.component.ts

```diff
@@ -15,12 +15,19 @@ import {GenericFormComponent} from '../../utils/miscellaneous/generic-form.compo
   styleUrls: ['./login.component.scss'],
   encapsulation: ViewEncapsulation.None
 })
-export class LoginComponent extends GenericFormComponent<SessionQuery, LoginFormState> {
+export class LoginComponent extends GenericFormComponent<SessionQuery, LoginFormState> implements OnInit {
   constructor(fm: AkitaNgFormsManager<FormState>,
               router: Router,
               private sessionService: SessionService,
-              private sessionQuery: SessionQuery) {
+              private sessionQuery: SessionQuery,
+              private _activatedRoute: ActivatedRoute
+  ) {
     super(fm, router, sessionQuery, 'login');
+
+    this._activatedRoute.queryParamMap
+      .subscribe((queryParams) => {
+        this.sessionService.backLink = queryParams.get('backLink');
```

Comment:
This is super minor, but our convention for query params is `snake_case`, so it's preferable for `queryParams.get('backLink')` to be `queryParams.get('back_link')`